src/state/store.ts

```diff
@@ -0,0 +1,28 @@
+import { Action, combineReducers, configureStore, Dispatch } from '@reduxjs/toolkit';
+import { legendReducer } from './legendSlice';
+
+const rootReducer = combineReducers({
+  // tooltip: tooltipReducer,
+  legend: legendReducer,
+});
+
+export const createStore = (chartName: string = 'Chart') => {
```

Comment:
This is a function because we want to allow multiple stores on one page. Users can choose to have multiple charts and they should be independent too.

The tooltip synchronisation will continue working the same as it does now - via the emitter.